apps/api/R/inputs.R

```diff
@@ -0,0 +1,116 @@
+library(dplyr)
+
+#' Search for Inputs containing wildcards for filtering
+#' @param model_id Model Id (character)
+#' @param site_id Site Id (character)
+#' @param offset
+#' @param limit 
+#' @return Information about Inputs based on model & site
+#' @author Tezan Sahu
+#* @get /
+searchInputs <- function(req, model_id=NULL, site_id=NULL, offset=0, limit=50, res){
+  if (! limit %in% c(10, 20, 50, 100, 500)) {
+    res$status <- 400
+    return(list(error = "Invalid value for parameter"))
+  }
+  
+  dbcon <- PEcAn.DB::betyConnect()
+  
+  inputs <- tbl(dbcon, "inputs") %>%
+    select(input_name=name, id, site_id, format_id, start_date, end_date)
+  
+  inputs <- tbl(dbcon, "dbfiles") %>%
+    select(file_name, file_path, container_type, id=container_id, machine_id) %>%
+    inner_join(inputs, by = "id") %>%
+    filter(container_type == 'Input') %>%
+    select(-container_type)
+  
+  inputs <- tbl(dbcon, "machines") %>%
+    select(hostname, machine_id=id) %>%
+    inner_join(inputs, by='machine_id') %>%
+    select(-machine_id)
+  
+  inputs <- tbl(dbcon, "modeltypes_formats") %>%
+    select(tag, modeltype_id, format_id, input) %>%
+    inner_join(inputs, by='format_id') %>%
+    filter(input) %>%
+    select(-input)
+  
+  inputs <- tbl(dbcon, "formats") %>%
+    select(format_id = id) %>%
+    inner_join(inputs, by='format_id') %>%
+    select(-format_id)
+  
+  inputs <- tbl(dbcon, "models") %>%
+    select(model_id = id, modeltype_id, model_name, revision) %>%
+    inner_join(inputs, by='modeltype_id') %>%
+    select(-modeltype_id)
+  
+  inputs <- tbl(dbcon, "sites") %>%
+    select(site_id = id, sitename) %>%
+    inner_join(inputs, by='site_id')
+  
+  if(! is.null(model_id)) {
```

Comment:
I think I'd argue that if you're not looking for a specific model_id that you don't even bother with the join to modeltypes_formats and models as there are definitely a lot of datasets in the Inputs table that are not associated with any specific model.

Reply:
But if we do that, we will lose the information about `tag`,  & `format_id` (which is needed for the join with the `formats` table).

Reply:
What do you mean? format_id is part of the Input record, so you should be joining `inputs.format_id = formats.id`. Not sure what `tag` is or why you'd need it. Indeed, if you only query formats that have matches in modeltype_formats then you're missing a LOT of formats.